### ISSUE TITLE
GPT-2

### ISSUE THREAD

**Issue by bipinkc19** [deepset-ai/FARM#492]
**Question**
I was wondering as one-shot learning of gpt-2 performs so well for it's type and gpt-3 has been released.
Is there any model Q/A model based on gpt-2 and if we can train it for the task?
 
**Additional context**
Very curious about it and I don't have that level of knowledge to guess it myself. It would be great if you guys could help me out.

**Comment by Timoeller** [comment]
Sure we can help out here.

So gpt-2 is a generative model, it could be used to generate answers just given the question. Bert or its descendants are only encoders (without the decoder to generate text), hence the name Bidirectional Encoder Representations from Transformers.
Most current QA methods - and all QA methods implemented in FARM - focus on extractive QA, that means there is question and text as input and the answer is extracted from this text (or more accurately, extracted from the encoded representation of the question and text).

Long story short, in FARM there are no gpt-2 like QA models and there are no immediate plans to implement them. You could use generative models to generate QA training data though. That is a use case that might be coming to FARM.

Hope this answer was helpful. Happy to elaborate more

**Comment by bipinkc19** [comment]
Thanks alot for the information. You guys are the best with such quick and informative replies every time.

These generative models, would they be good enough to produce specific questions to which answers are numbers. For example generating questions like "what is the growth rate this year?", whose answer is number. 
Do they generate answers in the label data too?

Could you point be to some of the model that is used for that

**Comment by Timoeller** [comment]
So I am not aware of a model that explicitly creates questions for numbers only, but it can be achieved by taking a normal question gen model and add some post processing to it - just take the questions whose answer is a number...

For question generation you might be interested in this thread: https://twitter.com/psuraj28/status/1290671279767351296?s=19

**Comment by bipinkc19** [comment]
Thanks a lot for the threads :)
